### PR TITLE
Transition to IPJ, Iroha rc2 version update

### PR DESCRIPTION
**Pull request by Mingela**
### Description of the Change
Now we don't need to support bindings. Just use and maintain IPJ library.
Iroha version was also updated to the recent release.

### CODE REVIEW

File: deploy/iroha/genesis.block [hunk 1]
```diff
@@ -264,11 +272,17 @@
                   "roleName": "registration_service"
                 }
               },
+              {
+                "appendRole": {
+                  "accountId": "btc_registration_service@notary",
+                  "roleName": "client"
```
Comment (dolgopolovwork) [inline deploy/iroha/genesis.block]: Looks suspicious. Why we use `client`?  Why not `registration_service`?

File: deploy/iroha/genesis.block [hunk 2]
```diff
@@ -284,11 +298,17 @@
                   "roleName": "registration_service"
                 }
               },
+              {
+                "appendRole": {
+                  "accountId": "mst_btc_registration_service@notary",
+                  "roleName": "client"
```
Comment (dolgopolovwork) [inline deploy/iroha/genesis.block]: Why `client`?

File: deploy/iroha/genesis.block [hunk 3]
```diff
@@ -342,11 +362,23 @@
                   "roleName": "tester"
                 }
               },
+              {
+                "appendRole": {
```
Comment (dolgopolovwork) [inline deploy/iroha/genesis.block]: Why we need this block? We have all desired permissions in `tester`

Comment (Mingela) [reply deploy/iroha/genesis.block]: These question can be answered by a single phrase: Iroha permissions and domains update.
Now a one need to have all the permissions and roles to issue a new account in domain with such role etc

**Pull request merged**